# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxUserFontSet.cpp
# Commit: e31c17c4b0a5
# Full Hash: e31c17c4b0a5056965d638894fae9b941c4b839b
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2021-04-08 09:51:11
# Description:
#   Bug 1703614 - Check for out-of-range source index when logging a font-load failure. r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D111153
# ==============================================================================

diff -r fa184a5e26d1 -r e31c17c4b0a5 gfx/thebes/gfxUserFontSet.cpp
--- a/gfx/thebes/gfxUserFontSet.cpp	Wed Apr 07 20:18:30 2021 +0000
+++ b/gfx/thebes/gfxUserFontSet.cpp	Wed Apr 07 20:26:02 2021 +0000
@@ -269,7 +269,7 @@
   aFamilyName = mFamilyName;
 
   aURI.Truncate();
-  if (aSrcIndex == mSrcList.Length()) {
+  if (aSrcIndex >= mSrcList.Length()) {
     aURI.AppendLiteral("(end of source list)");
   } else {
     if (mSrcList[aSrcIndex].mURI) {
